### PR TITLE
zephyr: disable clock before reset

### PR DESCRIPTION
**Pull request by rsalveti**
Avoid clock related interrupts before jumping into the first image. Safe
to disable as the first application will enable the clock/timer again
once booted.

Signed-off-by: Ricardo Salveti <ricardo.salveti@linaro.org>

### CODE REVIEW

**Comment by mike-scott** [comment]
When testing on Carbon, I put it in DFU mode, flash the new image and then swap the USB and it hangs at:
[MCUBOOT] [INF] main: Bootloader chainload address: 0x20000                                                                     
[MCUBOOT] [INF] main: Disabling clock and Jumping to the first image slot

**Comment by mike-scott** [comment]
Hm.. I was actually testing a rebase of upstream mcuboot master with this.  It may be due to one of the new commits.

More testing.

**Comment by mike-scott** [comment]
Yep it's working on our current branch for Carbon.   I'll hammer on Nitrogen a bit now.

**Review by mike-scott — APPROVED** [review]
LGTM.  Even if this change doesn't fix Nitrogen issue, this is probably needed.

**Comment by rsalveti** [comment]
This change doesn't completely fix the Nitrogen issue yet, but allows it to boot when building both the bootloader and the app with Zephyr 1.7-rc2.

File: boot/zephyr/main.c [hunk 1]
```diff
@@ -69,7 +70,8 @@ void main(void)
 	irq_lock();
 	_MspSet(vt->msp);
 
-	BOOT_LOG_INF("Jumping to the first image slot");
+	BOOT_LOG_INF("Disabling clock and Jumping to the first image slot");
+	sys_clock_disable();
```
Comment (mike-scott) [inline boot/zephyr/main.c:74]: What do you think about adding a "(void)irq_lock();" here similar to how sys_reboot() works: https://gerrit.zephyrproject.org/r/gitweb?p=zephyr.git;a=blob;f=misc/reboot.c;h=864aa0dd5ba7fe428e1531302df1b3800a3fa1c4;hb=refs/heads/master#l20

**Pull request merged**